src/clean-core/allocator.hh

```diff
@@ -142,55 +165,126 @@ struct stack_allocator final : allocator
     void reset()
     {
         _head = _buffer_begin;
-        _last_alloc = nullptr;
+        _last_alloc_id = 0;
     }
 
     stack_allocator() = default;
-    stack_allocator(span<std::byte> buffer) : _buffer_begin(buffer.data()), _head(buffer.data()), _buffer_end(buffer.data() + buffer.size()) {}
+    explicit stack_allocator(span<std::byte> buffer) : _buffer_begin(buffer.data()), _head(buffer.data()), _buffer_end(buffer.data() + buffer.size())
+    {
+    }
 
 private:
     std::byte* _buffer_begin = nullptr;
     std::byte* _head = nullptr;
     std::byte* _buffer_end = nullptr;
-    std::byte* _last_alloc = nullptr;
+    int32_t _last_alloc_id = 0;
```

Comment:
notice how this leaves a 32bit padding hole

Reply:
This was a bugfix not an optimization, the id is stored in alloc headers